### DISCUSSION TITLE
Please help - how to exclude steps from the "highlighting current step"

### DISCUSSION THREAD

**Discussion by forelockmocs** [trevorsandy/lpub3d#697] [Q&A] [answered]
I've recently finally updated my LPub3D to the latest version (2.4.6 build 3211). I cannot comprehend how to work with the meta-commands for the highlight settings in this new version. I am doing something wrong.

I want to enable to highlight the current step. But in the example I want to STOP the hightlighting after step 2 and start the hightlighting again at step 4 (so step 3 is NOT highlighted).

This my testcode - what am I doing wrong?

---
0 Name: Test
0 Author: LDraw
0 !LICENSE Redistributable under CCAL version 2.0 : see CAreadme.txt
0 !LPUB PAGE BACKGROUND GLOBAL COLOR "#d8effc"
0 !LPUB PAGE NUMBER PLACEMENT TOP_RIGHT PAGE INSIDE
0 !LPUB PAGE PAGE_HEADER SIZE GLOBAL 8.2677 0.3000
0 !LPUB RESOLUTION GLOBAL 300 DPI
0 !LPUB PAGE ORIENTATION GLOBAL LANDSCAPE
0 STEP
0 !LPUB ASSEM HIGHLIGHT_STEP GLOBAL SETUP TRUE
1 4 -51.932 21.064 0 1 0 0 0 1 0 0 0 1 3001.dat
0 STEP
1 14 -51.932 -2.936 0 1 0 0 0 1 0 0 0 1 3001.dat
0 STEP
0 !LPUB ASSEM HIGHLIGHT_STEP ENABLED FALSE
1 4 -51.932 -26.936 0 1 0 0 0 1 0 0 0 1 3001.dat
0 STEP
0 !LPUB ASSEM HIGHLIGHT_STEP ENABLED TRUE
1 14 -51.932 -50.936 0 1 0 0 0 1 0 0 0 1 3001.dat
0 STEP
0 // These 6 command lines were auto-generated for highlight current step.
0 // These lines are not saved and should not be modified, but they can be replaced.
0 // Remove comment lines and replace MODEL with DISPLAY_MODEL to override this behaviour.
0 !LPUB INSERT MODEL
0 !LPUB INSERT PAGE
0 STEP

**Comment by trevorsandy** [comment, marked as answer]
This should do the trick on the latest release build.

```
0 Name: Test
0 Author: LDraw
0 !LICENSE Redistributable under CCAL version 2.0 : see CAreadme.txt
0 !LPUB HIGHLIGHT_STEP SETUP GLOBAL TRUE
0 !LPUB HIGHLIGHT_STEP COLOR GLOBAL "#00ff00"
0 !LPUB HIGHLIGHT_STEP ENABLED GLOBAL FALSE
0 !LPUB PAGE BACKGROUND GLOBAL COLOR "#d8effc"
0 !LPUB PAGE NUMBER PLACEMENT TOP_RIGHT PAGE INSIDE
0 !LPUB PAGE PAGE_HEADER SIZE GLOBAL 8.2677 0.3000
0 !LPUB RESOLUTION GLOBAL 300 DPI
0 !LPUB PAGE ORIENTATION GLOBAL LANDSCAPE
0 // A STEP command after your global commands is necessary but as no parts are defined before it, your step numbering will not be affected.
0 STEP
0 !LPUB ASSEM HIGHLIGHT_STEP ENABLED TRUE
1 4 -51.932 21.064 0 1 0 0 0 1 0 0 0 1 3001.dat
0 STEP
1 14 -51.932 -2.936 0 1 0 0 0 1 0 0 0 1 3001.dat
0 STEP
0 !LPUB ASSEM HIGHLIGHT_STEP ENABLED FALSE
1 4 -51.932 -26.936 0 1 0 0 0 1 0 0 0 1 3001.dat
0 STEP
0 !LPUB ASSEM HIGHLIGHT_STEP ENABLED TRUE
1 14 -51.932 -50.936 0 1 0 0 0 1 0 0 0 1 3001.dat
0 STEP
0 // These 6 command lines were auto-generated for highlight current step.
0 // These lines are not saved and should not be modified, but they can be replaced.
0 // Remove comment lines and replace MODEL with DISPLAY_MODEL to override this behaviour.
0 !LPUB INSERT MODEL
0 !LPUB INSERT PAGE
0 STEP

```

Cheers,

**Reply by forelockmocs** [reply]
Thanks!!! Your support ROCKS! I had to put "ASSEM" in line 4, 5 and 6 in your "reply-code" to get it to work. Hope that's how you intended your code to work.

You have made one instruction maker very happy! 

```
0 Name: Test
0 Author: LDraw
0 !LICENSE Redistributable under CCAL version 2.0 : see CAreadme.txt
0 !LPUB ASSEM HIGHLIGHT_STEP SETUP GLOBAL TRUE
0 !LPUB ASSEM HIGHLIGHT_STEP COLOR GLOBAL "#00ff00"
0 !LPUB ASSEM HIGHLIGHT_STEP ENABLED GLOBAL FALSE
0 !LPUB PAGE BACKGROUND GLOBAL COLOR "#d8effc"
0 !LPUB PAGE NUMBER PLACEMENT TOP_RIGHT PAGE INSIDE
0 !LPUB PAGE PAGE_HEADER SIZE GLOBAL 8.2677 0.3000
0 !LPUB RESOLUTION GLOBAL 300 DPI
0 !LPUB PAGE ORIENTATION GLOBAL LANDSCAPE
0 // A STEP command after your global commands is necessary but as no parts are defined before it, your step numbering will not be affected.
0 STEP
0 !LPUB ASSEM HIGHLIGHT_STEP ENABLED TRUE
1 4 -51.932 21.064 0 1 0 0 0 1 0 0 0 1 3001.dat
0 STEP
1 14 -51.932 -2.936 0 1 0 0 0 1 0 0 0 1 3001.dat
0 STEP
0 !LPUB ASSEM HIGHLIGHT_STEP ENABLED FALSE
1 4 -51.932 -26.936 0 1 0 0 0 1 0 0 0 1 3001.dat
0 STEP
0 !LPUB ASSEM HIGHLIGHT_STEP ENABLED TRUE
1 14 -51.932 -50.936 0 1 0 0 0 1 0 0 0 1 3001.dat
0 STEP
0 // These 6 command lines were auto-generated for highlight current step.
0 // These lines are not saved and should not be modified, but they can be replaced.
0 // Remove comment lines and replace MODEL with DISPLAY_MODEL to override this behaviour.
0 !LPUB INSERT MODEL
0 !LPUB INSERT PAGE
0 STEP
```

**Comment by trevorsandy** [comment]
Excellent.

Cheers,
